### DISCUSSION TITLE
API to get or set any py-repl content

### DISCUSSION THREAD

**Discussion by WebReflection** [pyscript/pyscript#1339] [Proposals]
We've been recently reached out for help around the ability to store and restore python written within a `<py-repl>` custom element but unfortunately we don't have such functionality exposed so that I ended up suggesting [some sort of hack](https://gist.github.com/ntoll/b8f147bed9b20b39a001319dec12fddc?permalink_comment_id=4520880#gistcomment-4520880) to do so.

While the hack works pretty well with that specific use case, we agreed that it'd be great to have such functionality as `.value` accessor of the component, so that it's easy to retrieve all lines of code and set these back in the future whenever that's needed/desired.

Once agreed on the details we can move this into *issues* and I can help moving the proposal forward.

**Comment by JeffersGlass** [comment]
Love this idea! For what it's worth, this sounds similar to #954 - just referencing it here so it gets addressed/included/closed as appropriate in the process.

**Reply by WebReflection** [reply]
Thanks for the pointer ... I have tried without success mostly because I've looked for `py-repl` instead but yeah, it's a missing feature of the component that would probably be appreciated.

**Comment by JeffersGlass** [comment]
Does #1292 provide a satisfactory solution to this?

**Reply by WebReflection** [reply]
That helps but does not solve the stored use case (refresh and re-run previously edited code) but it could work with some indirection ... notably the data URL if it contains valid Python code so that could be saved in `localStorage` and retrieved/passed back on next run.

In short, to some extend, that helps with this issue to me so we can close this for the time being and re-open if needed in the future.

Thanks for the reminder.
